raphtory/src/db/api/view/edge_property_filter.rs

```diff
@@ -1,78 +1,207 @@
 
-    fn filter_edges_lt(&self, property: &str, value: Prop) -> Self::FilteredViewType {
-        let t_prop_id = self
-            .current_filter()
-            .edge_meta()
-            .get_prop_id(property, false);
-        let c_prop_id = self
-            .current_filter()
-            .edge_meta()
-            .get_prop_id(property, true);
+    fn filter_edges_lt(
+        &self,
+        property: &str,
+        value: Prop,
+    ) -> Result<Self::FilteredViewType, GraphError> {
+        let (t_prop_id, c_prop_id) =
+            get_ids_and_check_type(self.current_filter().edge_meta(), property, value.dtype())?;
+        Ok(self.one_hop_filtered(EdgePropertyFilteredGraph::new(
+            self.current_filter().clone(),
+            t_prop_id,
+            c_prop_id,
+            PropValueFilter::new(value, |left, right| left > right),
+        )))
+    }
+
+    fn filter_edges_le(
+        &self,
+        property: &str,
+        value: Prop,
+    ) -> Result<Self::FilteredViewType, GraphError> {
+        let (t_prop_id, c_prop_id) =
+            get_ids_and_check_type(self.current_filter().edge_meta(), property, value.dtype())?;
+        Ok(self.one_hop_filtered(EdgePropertyFilteredGraph::new(
+            self.current_filter().clone(),
+            t_prop_id,
+            c_prop_id,
+            PropValueFilter::new(value, |left, right| left >= right),
+        )))
+    }
+
+    fn filter_edges_eq(
+        &self,
+        property: &str,
+        value: Prop,
+    ) -> Result<Self::FilteredViewType, GraphError> {
+        let (t_prop_id, c_prop_id) =
+            get_ids_and_check_type(self.current_filter().edge_meta(), property, value.dtype())?;
+
+        Ok(self.one_hop_filtered(EdgePropertyFilteredGraph::new(
+            self.current_filter().clone(),
+            t_prop_id,
+            c_prop_id,
+            PropValueFilter::new(value, |left, right| left == right),
+        )))
+    }
+
+    fn filter_edges_ne(
+        &self,
+        property: &str,
+        value: Prop,
+    ) -> Result<Self::FilteredViewType, GraphError> {
+        let (t_prop_id, c_prop_id) =
+            get_ids_and_check_type(self.current_filter().edge_meta(), property, value.dtype())?;
+
+        Ok(self.one_hop_filtered(EdgePropertyFilteredGraph::new(
+            self.current_filter().clone(),
+            t_prop_id,
+            c_prop_id,
+            PropValueFilter::new(value, |left, right| left != right),
+        )))
+    }
+
+    fn filter_edges_gt(
+        &self,
+        property: &str,
+        value: Prop,
+    ) -> Result<Self::FilteredViewType, GraphError> {
+        let (t_prop_id, c_prop_id) =
+            get_ids_and_check_type(self.current_filter().edge_meta(), property, value.dtype())?;
+
+        Ok(self.one_hop_filtered(EdgePropertyFilteredGraph::new(
+            self.current_filter().clone(),
+            t_prop_id,
+            c_prop_id,
+            PropValueFilter::new(value, |left, right| left < right),
```

Comment:
Is there a reason these all read right to left?

Reply:
Swapped the arguments around so this is less confusing